plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-transact-account-manager.php

```diff
@@ -0,0 +1,381 @@
+	exit;
+}
+
+/**
+ * Handles Transact account management.
+ */
+final class WC_Gateway_Paypal_Transact_Account_Manager {
+	/**
+	 * The API version for the proxy endpoint.
+	 *
+	 * @var int
+	 */
+	private const WPCOM_PROXY_ENDPOINT_API_VERSION = 2;
+
+	/**
+	 * Transact provider type, for provider onboarding.
+	 *
+	 * @var string
+	 */
+	private const TRANSACT_PROVIDER_TYPE = 'paypal_standard';
+
+	/**
+	 * Cache keys for the merchant and provider accounts.
+	 *
+	 * @var string
+	 */
+	private const TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_LIVE = 'woocommerce_paypal_transact_merchant_account_live';
+	private const TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_TEST = 'woocommerce_paypal_transact_merchant_account_test';
+	private const TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_LIVE = 'woocommerce_paypal_transact_provider_account_live';
+	private const TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_TEST = 'woocommerce_paypal_transact_provider_account_test';
+
+	/**
+	 * The expiry time for the Transact account cache.
+	 *
+	 * @var int
+	 */
+	private const TRANSACT_ACCOUNT_CACHE_EXPIRY = 60 * 60 * 24; // 24 hours.
+
+	/**
+	 * Paypal gateway object.
+	 *
+	 * @var WC_Gateway_Paypal
+	 */
+	private $gateway;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Gateway_Paypal $gateway Paypal gateway object.
+	 */
+	public function __construct( $gateway ) {
+		$this->gateway = $gateway;
+	}
+
+	/**
+	 * Onboard the merchant with the Transact platform.
+	 *
+	 * @return void
+	 */
+	public function do_onboarding() {
+		// Check that we have a PayPal email. This is required for processing payments --
+		// used as the payee email. Only begin onboarding if this minimum requirement is met.
+		if ( empty( $this->gateway->email ) ) {
+			return;
+		}
+
+		// Register with Jetpack if not already connected.
+		$jetpack_connection_manager = $this->gateway->get_jetpack_connection_manager();
+		if ( ! $jetpack_connection_manager ) {
+			WC_Gateway_Paypal::log( 'Jetpack connection manager not found.', 'error' );
+			return;
+		}
+
+		if ( ! $jetpack_connection_manager->is_connected() ) {
+			$result = $jetpack_connection_manager->try_registration();
+			if ( is_wp_error( $result ) ) {
+				WC_Gateway_Paypal::log( 'Jetpack registration failed: ' . $result->get_error_message(), 'error' );
+				return;
+			}
+		}
+
+		// Fetch (cached) or create the Transact merchant and provider accounts.
+		$merchant_account_data = $this->get_merchant_account_data();
+		if ( empty( $merchant_account_data ) ) {
+			$merchant_account = $this->create_merchant_account();
+			if ( empty( $merchant_account ) ) {
+				WC_Gateway_Paypal::log( 'Transact merchant onboarding failed.', 'error' );
+				return;
+			}
+		}
+
+		$provider_account_data = $this->get_provider_account_data();
+		if ( empty( $provider_account_data ) ) {
+			$provider_account = $this->create_provider_account();
+			if ( ! $provider_account ) {
+				WC_Gateway_Paypal::log( 'Transact provider onboarding failed.', 'error' );
+				return;
+			}
+		}
```

Comment:
Suggestion: after creating the merchant and provider accounts, we can cache them immediately if the request is successful.